core/src/main/java/org/apache/calcite/util/interval/SparkDateTimestampInterval.java

```diff
@@ -67,12 +67,13 @@ public boolean unparseDateTimeMinus(SqlWriter writer, SqlCall call,
       case "INTERVAL_MONTH":
         handleIntervalMonth(writer, call, leftPrec, rightPrec, sign);
         break;
-      case "INTERVAL_DAY":
```

Comment:
Why this change is required ?

Reply:
This was actually overriding the unparse method of DATE_ADD, DATE_SUB operators so it was initially removed, but since TIMESTAMP_ADD, TIMESTAMP_SUB operators are coming to this switch statement so added it back and written DATE_ADD, DATE_SUB specific logic instead.